ycbrcr.go

```diff
@@ -0,0 +1,143 @@
+// +build !go1.5
+
+package resize
+
+import "image"
+
+// YCbCr converts ycc to a YCbCr image with the same subsample ratio
+// as the YCbCr image that ycc was generated from.
+func (p *ycc) YCbCr() *image.YCbCr {
+	ycbcr := image.NewYCbCr(p.Rect, p.SubsampleRatio)
+	var off int
+
+	switch ycbcr.SubsampleRatio {
+	case image.YCbCrSubsampleRatio422:
+		for y := ycbcr.Rect.Min.Y; y < ycbcr.Rect.Max.Y; y++ {
```

Comment:
How about extracting the codes in the `case`s into functions and put them into another file? This could avoid some code duplication: One file provides the functions for all Go versions, another file the functions for Go1.5. `ycbrcr.go` and `ycbrcr_go15.go` will be much smaller.